processor/deltatorateprocessor/config.go

```diff
@@ -26,6 +27,41 @@ type Config struct {
 
 	// List of delta sum metrics to convert to rates
 	Metrics []string `mapstructure:"metrics"`
+
+	// Time unit to calculate the rate over. This could be one of [ns, ms, s, m]. Default is s.
+	TimeUnit StringTimeUnit `mapstructure:"time_unit"`
```

Comment:
Is rate usually per second? Can we not have this for the moment?

Reply:
For CPU usage are recorded in nanoseconds and those are calculated over Nanoseconds in our default EKS Container Insights. But for other metrics like error count, network bytes are usually per second. To me it seems like good to have.

Reply:
I am lost, can you explain? For CPU-usage that is reported as 300ns (delta), the rate will be 300/(end-starttime in seconds)

Reply:
If you want to get CPU seconds/second then actually will be strange to do this in one op. So need to convert the metric to seconds first.

Reply:
I agree. If we need to convert the metrics, we can do that in an earlier stage using the metrics_geneartion processor scaling operation. 

For the rate calculations, we are giving customers the option. They knows which metrics are being recorded in what units. Say
1. CPU metric unit is in ns and time units are in ns. This one should be usage_ns/ time_difference_ns
1. Error Count metric unit in count and time units are in ns. For this one, maybe customer wants to calculate it per second.

Reply:
I am not sure if the above mentioned use cases are convincing enought to have this option. Plese, let me know your opinion.

Reply:
> CPU metric unit is in ns and time units are in ns. This one should be usage_ns/ time_difference_ns

I can see how this time unit can work if the metric also has a unit time but is a very special case that we should probably  not worry about it in this processor.

> Error Count metric unit in count and time units are in ns. For this one, maybe customer wants to calculate it per second.

Which time units are in `ns`?

Reply:
> Which time units are in `ns`?

`dp.StartTimestamp` and `dp.Timestamp` both are in `ns`, right?

Reply:
So whats your final suggestion? We will only support the default rate calculation per/second?

Reply:
> dp.StartTimestamp and dp.Timestamp both are in ns, right? indeed and always are like that. So you always have to transform them to second.

Reply:
Thanks. I will remove the config option for now and make the default rate calculation per/second.

Reply:
waiting for the change :)

Reply:
Pushed an update. Thanks.